.github/workflows/release.yml

```diff
@@ -0,0 +1,163 @@
+          IS_PRERELEASE=$(jq --raw-output ".announcement_is_prerelease" dist-manifest.json)
+          jq --raw-output ".announcement_github_body" dist-manifest.json > new_dist_announcement.md
+          gh release create ${{ github.ref_name }} --draft --prerelease="$IS_PRERELEASE" --title="$ANNOUNCEMENT_TITLE" --notes-file=new_dist_announcement.md
+          echo "created announcement!"
+
+          # Upload the manifest to the Github Release™
+          gh release upload ${{ github.ref_name }} dist-manifest.json
+          echo "uploaded manifest!"
+
+          # Disable all the upload-artifacts tasks if we have no actual releases
+          HAS_RELEASES=$(jq --raw-output ".releases != null" dist-manifest.json)
+          echo "has-releases=$HAS_RELEASES" >> "$GITHUB_OUTPUT"
+
+  # Build and packages all the things
+  upload-artifacts:
+    # Let the initial task tell us to not run (currently very blunt)
+    needs: create-release
+    if: ${{ needs.create-release.outputs.has-releases == 'true' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # For these target platforms
+        include:
+        - os: "ubuntu-20.04"
+          dist-args: "--artifacts=local --target=x86_64-unknown-linux-musl"
+          install-dist: |
+            cargo install \
+              --git "https://github.com/webern/cargo-dist" \
+              --rev "3dcbe823db0c4068f38db790d996a4af54e68b3e"
+        - os: "ubuntu-20.04"
+          dist-args: "--artifacts=local --target=aarch64-unknown-linux-musl"
+          install-dist:  |
+            cargo install \
+              --git "https://github.com/webern/cargo-dist" \
+              --rev "3dcbe823db0c4068f38db790d996a4af54e68b3e"
+    runs-on: ${{ matrix.os }}
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      RUSTC_BOOTSTRAP: "1"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install cargo-dist
+        run: ${{ matrix.install-dist }}
+      - name: Install Cross
+        run: |
+          cargo install cross --git https://github.com/cross-rs/cross
+      - name: Default Rust to Nightly
+        run: |
+          rustup default nightly
+      - name: Run cargo-dist
+        # This logic is a bit janky because it's trying to be a polyglot between
+        # powershell and bash since this will run on windows, macos, and linux!
+        # The two platforms don't agree on how to talk about env vars but they
+        # do agree on 'cat' and '$()' so we use that to marshal values between commands.
+        #
+        # We also have to do acrobatics to support cross-compilation to arm.
+        # Adapted from https://www.modio.se/cross-compiling-rust-binaries-to-armv7.html
+        run: |
+          export PATH=$CARGO_HOME/bin:$PATH
+          # Actually do builds and make zips and whatnot
+          cargo dist build --verbose=debug --tag=${{ github.ref_name }} --output-format=json ${{ matrix.dist-args }} > dist-manifest.json
```

Comment:
I think it would help future rebases to split this into two commits:
* one for the unmodified `cargo dist` workflow
* one for the local changes we've made to that workflow

That way, the local changes can be picked out and potentially reapplied if we want to rebase to a newer cargo dist release for new features or something.